### PR TITLE
Consistent naming for Validation, Either and Disjunction.

### PR DESCRIPTION
**Pull request by jozi-k**

### CODE REVIEW

File: core/src/main/scala/scalaz/syntax/std/BooleanOps.scala [hunk 1]
```diff
@@ -282,7 +282,10 @@ final class BooleanOps2(self: Boolean) {
    * Returns the first argument in `\/-` if this is `true`, otherwise the second argument in
    * `-\/`.
    */
-  final def either[A](a: => A): ConditionalEither[A] = new ConditionalEither(a)
+  final def toEither[A](a: => A): ConditionalEither[A] = new ConditionalEither(a)
+
+  @deprecated("Use `toEither`", "7.3.0")
+  final def either[A](a: => A): ConditionalEither[A] = toEither(a)
```
Comment (joroKr21) [inline core/src/main/scala/scalaz/syntax/std/BooleanOps.scala]: This one is a bit special. It's used as `bool.either(right).or(left)` so I don't think it should be renamed.

**Comment by joroKr21** [comment]
I would also be interested to know if a PR deprecating `Maybe.cata` in favour of `fold` would be accepted :smile: 

It's been a thorn in my eyes for a long time.

File: core/src/main/scala/scalaz/syntax/std/BooleanOps.scala (hunk 1 above)
Comment (jozi-k) [reply core/src/main/scala/scalaz/syntax/std/BooleanOps.scala]: nice catch @joroKr21 !

**Comment by jozi-k** [comment]
I think you should ask at gitter channel about the `Maybe.cata` or just open PR and see what happens :)

**Comment by tonymorris** [comment]
The original function was called `fold` before the stdlib started taking that name, and so was renamed to `cata` so that any implicit conversion had access to that function. An overall view of naming, such that it is consistent, and unlikely to clash with a future stdlib (unicode? :)) is definitely warranted.

**Comment by jozi-k** [comment]
Can someone from maintainers have a brief look at this PR please?

File: core/src/main/scala/scalaz/syntax/std/EitherOps.scala [hunk 2]
```diff
@@ -6,7 +6,10 @@ final class EitherOps[A, B](private val self: Either[A, B]) extends AnyVal {
 
   final def disjunction: A \/ B = \/ fromEither self
```
Comment (alexknvl) [inline core/src/main/scala/scalaz/syntax/std/EitherOps.scala]: Shouldn't this one also be `toDisjunction`?

Comment (jozi-k) [reply core/src/main/scala/scalaz/syntax/std/EitherOps.scala]: Great observation! I added it to newest version of this PR.

**Comment by jozi-k** [comment]
PR has been updated. Asking for another round of reviews :)

**Pull request merged**